src/Microsoft.Management.Deployment/PackageCatalogReference.cpp

```
@@ -1,97 +1,121 @@
+                ::AppInstaller::Repository::SourceDetails sourceDetails = catalogInfoImpl->GetSourceDetails();
+                source = ::AppInstaller::Repository::OpenSourceFromDetails(sourceDetails, progress).Source;
+            }
+
+            if (!source)
+            {
+                // If source is null, return the error. There's no way to get the hresult that caused the error right now.
+                auto connectResult = winrt::make_self<wil::details::module_count_wrapper<winrt::Microsoft::Management::Deployment::implementation::ConnectResult>>();
+                connectResult->Initialize(winrt::Microsoft::Management::Deployment::ConnectResultStatus::CatalogError, nullptr);
+                return *connectResult;
+            }
+
+            // Have to make another package catalog info because source->GetDetails has more fields than m_info does.
+            // Specifically, Rest sources do not have the Ids filled in m_info since they only get the id from the rest server after being Opened.
+            auto packageCatalogInfo = winrt::make_self<wil::details::module_count_wrapper<winrt::Microsoft::Management::Deployment::implementation::PackageCatalogInfo>>();
+            packageCatalogInfo->Initialize(source->GetDetails());
+            auto connectResult = winrt::make_self<wil::details::module_count_wrapper<winrt::Microsoft::Management::Deployment::implementation::ConnectResult>>();
+            auto packageCatalog = winrt::make_self<wil::details::module_count_wrapper<winrt::Microsoft::Management::Deployment::implementation::PackageCatalog>>();
+            packageCatalog->Initialize(*packageCatalogInfo, source, (m_compositePackageCatalogOptions != nullptr));
+            connectResult->Initialize(winrt::Microsoft::Management::Deployment::ConnectResultStatus::Ok, *packageCatalog);
+            return *connectResult;
+        }
+        catch (const wil::ResultException&)

```

Do you plan on doing anything with these individual exception types?  If you want to just swallow all exceptions the last one is sufficient.

---

Yeah, thats true. I added them thinking I'd add logging before I realized the hresult can't be set anywhere and its already being logged. Removed.